evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py

```diff
@@ -217,8 +239,9 @@ def transform(
                     index=truncated_y.index,
                 ),
             )
-
-        return X, y_in_sample.append(y_out_of_sample)
+        y_t = y_in_sample.append(y_out_of_sample)
+        y_t.index = original_index
```

Comment:
minor point: should we use `self.original_index` here for clarity? or set `self.original_index lower`?